src/Model/__tests__/Model.test.ts

```diff
@@ -0,0 +1,111 @@
+import { FakeLogger } from '@adonisjs/logger/build/standalone';
+
+import { Mongodb } from '../../Mongodb';
+import { Model } from '../Model';
+
+interface IUser {
+  username: string;
+  password: string;
+}
+
+class User extends Model implements IUser {
+  public static collectionName = 'users';
+
+  public username: string;
+  public password: string;
+}
+
+const loggerConfig = {
+  name: 'adonis-logger',
+  level: 'trace',
+  messageKey: 'msg',
+  enabled: true,
+};
+const mongoConfig = {
+  default: 'mongo',
+  connections: {
+    mongo: {
+      url: 'mongodb://127.0.0.1:33333',
+      database: 'test-runner',
+    },
+  },
+};
+const db = new Mongodb(mongoConfig, new FakeLogger(loggerConfig));
+Model.$setDatabase(db);
+
+afterAll(async () => {
+  await (await User.getCollection()).deleteMany({});
+  await db.closeConnections();
+});
+
+test('can create', async () => {
+  const newUser = await User.create({
+    username: 'root',
+    password: 'root',
+  });
+  expect(newUser).toBeDefined();
+});
+
+test('get collection by model class', async () => {
+  const collection = await User.getCollection();
+  expect(collection).toBeDefined();
+});
+
+test('find one by property', async () => {
+  const user = User.findOne({ username: 'root', password: 'root' });
+  expect(user).toBeDefined();
+});
+
+test('find all', async () => {
+  await User.create({
+    username: 'root2',
+    password: 'root',
+  });
+
+  const users = await User.find({});
+  expect(await users.count()).toBe(2);
+});
+
+test('find by id should work', async () => {
+  const user = await User.create({
+    username: 'root3',
+    password: 'root',
+  });
+  const secondUser = await User.findById(user.id);
+  expect(secondUser).not.toBeNull();
+});
+
+test("find by id should throw when doesn't exists", async () => {
+  const t = async () => {
+    await User.findByIdOrThrow('notavalidid');
+  };
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises, jest/valid-expect
+  expect(t).rejects.toStrictEqual(
+    new Error('document notavalidid not found in users'),
+  );
+});
+
+test('saved changes should be sent to database', async () => {
+  const user = await User.create({
+    username: 'root4',
+    password: 'root',
+  });
+  user.password = 'rootroot';
```

Comment:
You could check that the user's id is a ObjectId

Reply:
Done